File: mycollab-services-community/src/test/java/com/mycollab/module/project/service/ProjectNotificationServiceTest.java
package com.mycollab.module.project.service;

import com.mycollab.module.project.domain.ProjectRelayEmailNotification;
import com.mycollab.test.DataSet;
import com.mycollab.test.spring.IntegrationServiceTest;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

@RunWith(SpringJUnit4ClassRunner.class)
public class ProjectNotificationServiceTest extends IntegrationServiceTest {
    @Autowired
    private ProjectService projectService;

    @DataSet
    @Test
    public void testGetNotification() {
        List<ProjectRelayEmailNotification> projectRelayEmailNotifications = projectService.findProjectRelayEmailNotifications();
        assertThat(projectRelayEmailNotifications.size()).isEqualTo(1);
        ProjectRelayEmailNotification projectRelayEmailNotification = projectRelayEmailNotifications.get(0);
        assertThat(projectRelayEmailNotification.getNotifyUsers().size()).isEqualTo(2);
        assertThat(projectRelayEmailNotification.getNotifyUsers()).extracting("username").
                contains("admin").contains("user2");
    }
}

File: mycollab-services-community/src/test/java/com/mycollab/module/project/service/ProjectNotificationServiceTest.kt
package com.mycollab.module.project.service

import com.mycollab.test.DataSet
import com.mycollab.test.spring.IntegrationServiceTest
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import org.junit.runner.RunWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.test.context.junit4.SpringJUnit4ClassRunner

@RunWith(SpringJUnit4ClassRunner::class)
class ProjectNotificationServiceTest : IntegrationServiceTest() {
    @Autowired
    private lateinit var projectService: ProjectService

    @DataSet
    @Test
    fun testGetNotification() {
        val projectRelayEmailNotifications = projectService.findProjectRelayEmailNotifications()
        assertThat(projectRelayEmailNotifications.size).isEqualTo(1)
        val projectRelayEmailNotification = projectRelayEmailNotifications[0]
        assertThat(projectRelayEmailNotification.notifyUsers.size).isEqualTo(2)
        assertThat(projectRelayEmailNotification.notifyUsers).extracting("username").contains("admin").contains("user2")
    }
}
